components/supervisor/pkg/supervisor/tasks.go

```diff
@@ -205,7 +205,7 @@ func (tm *tasksManager) init(ctx context.Context) {
 	}
 }
 
-func (tm *tasksManager) Run(ctx context.Context, wg *sync.WaitGroup) {
+func (tm *tasksManager) Run(ctx context.Context, wg *sync.WaitGroup, successChan chan bool) {
```

Comment:
nit: instead of passing this in, you could return `success <-chan bool`. This way the only writer to the channel is sure to be in this function,

Reply:
This func is meant be called as goroutine and wire it up before it returns (to keep it in-style with the other methods).